### ISSUE TITLE
Search engine results do not link to the corresponding detail page

### ISSUE THREAD

**Issue by patrickakk** [acdh-oeaw/dhcr-main#55]
search result for a specific course links to main page, instead of to the course detail page

**Comment by dietervu** [comment]
For completeness as screenshot: 
<img width="883" alt="image" src="https://user-images.githubusercontent.com/8073487/216378070-5a586fd6-718b-4fe8-b068-6c9319b5f9e5.png">

source: English google, search URL: https://www.google.com/search?client=firefox-b-d&q=Digital+Humanities+Across+Borders+course

**Comment by patrickakk** [comment]
The problem above could maybe be solved by creating a sitemap.

As preparation:
Today I've removed the word "Sitemap" from both menu's in the public part and login area, since those menu's are not a sitemap. The word has been replaced by the logo.

Does everybody agree with the new layout ?
### Public menu - before
![image](https://user-images.githubusercontent.com/97846194/233842880-9ae80b90-39a2-447f-8aa5-18ced08342a4.png)

### Public menu - after
![image](https://user-images.githubusercontent.com/97846194/233842962-17e8ebf3-9a2f-4bc1-aacb-b80a92e2bf6d.png)

### Login area menu - before
![image](https://user-images.githubusercontent.com/97846194/233842897-033fd430-3687-4e64-b9bb-cb6c05cdecc0.png)

### Login area menu - after
![image](https://user-images.githubusercontent.com/97846194/233842979-ebfb74d4-7842-43d7-8845-7e8fd93c970a.png)

**Comment by patrickakk** [comment]
The change above is implemented in 2023-04. Currently blocked until 2023-03 is released. Then is can be reviewed.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer Can you review this?

**Comment by PixlTracer** [comment]
I am ok with it!

**Comment by patrickakk** [comment]
@PixlTracer Ok, thanks for the reply. Then I'll move the issue back to the jul23 milestone for the original task.

@IvdL22 Please reply if you have other thoughts about the layout change.

**Comment by patrickakk** [comment]
Proposed solution:
First create an xml sitemap and list that in robots.txt.
In case the search engine results don't improve, we could look at other options.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 

Can you review this?

I've create a tool that lists **all public shown courses** together with their last updated date in the sitemap. That are currently 192 urls. As well some static pages are listed. 

**Which static pages do we want listed in the sitemap = be found in a search engine?**
I've added the following urls:
- https://dhcr.clarin-dariah.eu
- https://dhcr.clarin-dariah.eu/info
- https://dhcr.clarin-dariah.eu/info#purpose
- https://dhcr.clarin-dariah.eu/national-moderators
- https://dhcr.clarin-dariah.eu/info#publications
- https://dhcr.clarin-dariah.eu/info#data
- https://dhcr.clarin-dariah.eu/info#clarin-dariah
- https://dhcr.clarin-dariah.eu/faq/public

Do we need to add or remove pages from that list?

The sitemap in xml can be found here:
https://dev-dhcr.clarin-dariah.eu/sitemap.xml

Can you change the label to Done if you're satisfied?

```
Note for myself: Todo after review/release:
- Add to robots.txt
- Create daily cronjob
```

**Comment by PixlTracer** [comment]
elegant solution, thanks for the suggestions!
I think this list should be sufficient... I guess, we could add additional links to it in case we might come across some other URL to include?
I'll set the label to done

**Comment by patrickakk** [comment]
@PixlTracer 
Yes links to static pages can be added in the future. (Please mind that this is described in the code, so changes need time to a the next release.)

For this issue, I'l take on the rest now, todo:

- [x] Enable sitemap generation in production
- [x] Add robot.txt to make sitemap findable
- [x] Release version
- [x] Add cronjob for regenerating sitemap every day (24 hours)
- [x] Block this issue for 6 weeks
- [ ] After that: Check results in multiple search engines and determine necessary actions, first url to check: https://www.google.com/search?client=firefox-b-d&q=Digital+Humanities+Across+Borders+course

**Comment by patrickakk** [comment]
@PixlTracer 
I've removed https://dhcr.clarin-dariah.eu/national-moderators from the sitemap, since this would also crawl and save the moderators email addresses. Please let me know if you prefer otherwise.

**Comment by patrickakk** [comment]
Blocked until 2023-09-13 (6 weeks from now).

Then test listing in search engines.

**Comment by patrickakk** [comment]
Currently waiting for availability of Google Webmaster Console.

**Comment by patrickakk** [comment]
Update: 
Google Webmaster Console is available since today.

**Comment by patrickakk** [comment]
Status update:
To be discussed in January meeting

**Comment by vronk** [comment]
As discussed: Let's try to improve the indexing via Google search console.
providing the sitemap.xml

**Comment by dietervu** [comment]
On Firefox the URLs that sitemap.xml links to give intermittent errors, e.g. https://dhcr.clarin-dariah.eu/courses/view/804 results in a semi-empty page:

![image](https://github.com/user-attachments/assets/6363a5ca-1c55-47f8-b6ce-9dea513429f7)

The console looks then like this:

![image](https://github.com/user-attachments/assets/c541ae60-6697-4cb5-9ab1-9aca092a9271)

It might be that the google crawler is experiencing something similar?

**Comment by patrickakk** [comment]
For info, the sitemap can be found here:
https://dhcr.clarin-dariah.eu/sitemap.xml

**Comment by patrickakk** [comment]
@dietervu 
Thanks for the bug report, I've split that into a new issue, here:
https://github.com/acdh-oeaw/dhcr-main/issues/127

**Comment by patrickakk** [comment]
Status update:
Currently priority is given to other issue.

**Comment by patrickakk** [comment]
Info: 
For questions about the indexing, Twan can be asked.

Problems with the course detail page are also addressed in #127
